### PR TITLE
update to static signals

### PR DESCRIPTION
**Pull request by sccolbert**
pinging @dwillmer @blink1073 @ellisonbg @sylvaincorlay @jasongrout

Source, tests, and API docs are updated - working on the README now.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -11,36 +11,106 @@
...
+
+/**
+ * A typedef for a signal callback function.
+ *
+ * @param T - The type of the sender.
+ *
+ * @param U - The type of the signal args.
+ */
+export
+type Slot<T, U> = (sender: T, args: U) => void;
```
Comment (jasongrout) [inline src/index.ts]: I thought we were going to make the sender the optional second argument.

Comment (sccolbert) [inline src/index.ts]: I didn't think it made sense to reverse the order of types for this one thing. I've come across libraries which did that before, and I remember it hating the inconsistency.

Also, precedent: https://msdn.microsoft.com/en-us/library/bb655886(v=vs.90).aspx

**Comment by sccolbert** [comment]
This is blocking me from publishing more widgets, so I'd like to get y'all's +1's asap.

File: src/index.ts (hunk 1 above)
Comment (jasongrout) [inline src/index.ts]: Another: https://docs.djangoproject.com/en/1.8/topics/signals/#receiver-functions

Comment (sccolbert) [inline src/index.ts]: Nice find. Didn't know Django had such functionality.

**Comment by dwillmer** [comment]
+1

**Comment by ellisonbg** [comment]
+1

On Fri, Sep 4, 2015 at 1:41 PM, Dave Willmer <notifications@github.com>
wrote:

> +1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phosphorjs/phosphor-signaling/pull/6#issuecomment-137847344>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by SylvainCorlay** [comment]
+1 and it is closer to your very first version for unbound signals.

**Comment by blink1073** [comment]
:+1:

**Comment by jasongrout** [comment]
+1

**Comment by sccolbert** [comment]
Thanks gents!

**Comment by sccolbert** [comment]
>  and it is closer to your very first version for unbound signals.

Yep. This has proven to be a tricky one to get right.

**Comment by jasongrout** [comment]
@SylvainCorlay is glad he waited all this time to update things using signals :).

**Pull request merged**